dvc/repo/brancher.py

```diff
@@ -34,44 +32,34 @@ def brancher(  # noqa: E302
     from dvc.fs.local import LocalFileSystem
 
     saved_fs = self.fs
-    revs = revs.copy() if revs else []
 
     scm = self.scm
 
     self.fs = LocalFileSystem(url=self.root_dir)
     yield "workspace"
 
-    if revs and "workspace" in revs:
+    revs = revs.copy() if revs else []
+    if "workspace" in revs:
         revs.remove("workspace")
 
-    if all_commits:
-        revs = scm.list_all_commits()
-    else:
-        if all_branches:
-            revs.extend(scm.list_branches())
-
-        if all_tags:
-            revs.extend(scm.list_tags())
-
-    if all_experiments:
-        from dvc.repo.experiments.utils import exp_commits
-
-        revs.extend(exp_commits(scm))
+    found_revs = iter_revs(
+        scm,
+        revs,
+        all_branches=all_branches,
+        all_tags=all_tags,
+        all_commits=all_commits,
+        all_experiments=all_experiments,
+    )
 
     try:
-        if revs:
-            from dvc.fs.git import GitFileSystem
-            from dvc.scm import resolve_rev
+        if found_revs:
+            for sha, names in found_revs.items():
```

Comment:
This `if` isn't needed

```suggestion
        for sha, names in found_revs.items():
```